src/Spout/Reader/ODS/RowIterator.php

```diff
@@ -99,15 +108,34 @@ public function valid()
      */
     public function next()
     {
+        $prevRow = null;
+
+        if (count($this->rowDataBuffer) > 1) {
```

Comment:
I agree it's too intricate, I'll investigate if it's possible to use a solution where only one row is buffered (which would also address your other comment regarding the repeated rows).